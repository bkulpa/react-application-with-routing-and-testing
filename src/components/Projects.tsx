import { FC } from 'react';
import { ProjectsList } from './ProjectsList';

export const Projects: FC = () => {
    return (
        <>
            <h2>Check our Projects!</h2>
            <img className='img__about' src={require("../img/games.jpg")} alt={"Games alphabet"} />
            <ul>
                {ProjectsList.map(({ projectName, score }) => (
                    <li key={projectName}>Game {projectName} with score: {score}!</li>
                ))}
            </ul>
        </>
    )
}