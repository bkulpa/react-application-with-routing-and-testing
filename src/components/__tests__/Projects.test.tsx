import { render } from '@testing-library/react';
import { Projects } from '../Projects';

describe('Projects', () => {
    it('should render', () => {
        // before
        const { asFragment } = render(<Projects />);
        // result
        expect(asFragment()).toMatchSnapshot();
    });
});