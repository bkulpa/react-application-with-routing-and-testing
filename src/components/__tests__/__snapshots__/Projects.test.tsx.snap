// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should render 1`] = `
<DocumentFragment>
  <h2>
    Check our Projects!
  </h2>
  <img
    alt="Games alphabet"
    class="img__about"
    src="games.jpg"
  />
  <ul>
    <li>
      Game Witcher with score: 99!
    </li>
    <li>
      Game Mortal Combat with score: 70!
    </li>
    <li>
      Game Fifa 2022 with score: 1!
    </li>
    <li>
      Game Assetto Corsa with score: 87!
    </li>
  </ul>
</DocumentFragment>
`;
